### PR TITLE
Fixed flakey test from RandomSourceTest as the binary search needed a ceil to find the first match in the list.  Fixed a bug where the seed can get clost or put into the wrong test

### PR DESCRIPTION
**Pull request by dcapwell**

### CODE REVIEW

File: accord-core/src/test/java/accord/utils/RandomSourceTest.java [hunk 1]
```diff
@@ -111,18 +104,101 @@ private double testOneBiasedLongs(RandomSource random, int min, int median, int
...
-        int i = Arrays.binarySearch(results, median);
+        int i = firstBinarySearch(results, median);
         if (i < 0) i = -1 - i;
-        int j = Arrays.binarySearch(results, median + 1);
+        int j = firstBinarySearch(results, median + 1);
         if (j < 0) j = -2 - j;
         else --j;
-        i -= results.length/2;
-        j -= results.length/2;
+        i -= Math.abs(results.length/2);
```
Comment (belliottsmith) [inline accord-core/src/test/java/accord/utils/RandomSourceTest.java]: I don't think this is necessary? It's not possible for `results.length/2` to be negative?

File: accord-core/src/test/java/accord/utils/RandomSourceTest.java [hunk 2]
```diff
@@ -111,18 +104,101 @@ private double testOneBiasedLongs(RandomSource random, int min, int median, int
             results[i] = random.nextBiasedInt(min, median, max);
 
         Arrays.sort(results);
-        int i = Arrays.binarySearch(results, median);
+        int i = firstBinarySearch(results, median);
         if (i < 0) i = -1 - i;
-        int j = Arrays.binarySearch(results, median + 1);
+        int j = firstBinarySearch(results, median + 1);
```
Comment (belliottsmith) [inline accord-core/src/test/java/accord/utils/RandomSourceTest.java]: If we're copying `ceil`, we could just use `ceil` directly?

Comment (dcapwell) [reply accord-core/src/test/java/accord/utils/RandomSourceTest.java]: renamed the method to `ceil`. I also simplified it to call the forked `ceil` and only exists to simplify the signature.

**Comment by belliottsmith** [comment]
+1

**Pull request merged**